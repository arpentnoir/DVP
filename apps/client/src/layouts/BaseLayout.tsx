import { Container, ContainerProps } from '@mui/material';
import { Helmet } from 'react-helmet';

interface IBaseLayout extends ContainerProps {
  children: React.ReactNode;
  title: string;
}

export const BaseLayout = ({ children, title, ...rest }: IBaseLayout) => {
  return (
    <>
      <Helmet>
        <title>{title}</title>
      </Helmet>

      <Container
        maxWidth={false}
        {...rest}
        sx={{
          paddingX: { xs: '40px', lg: '0' },
          paddingY: { xs: '40px', md: '60px' },
          maxWidth: '950px',
          flex: 1,
          ...rest.sx,
        }}
        disableGutters
      >
        {children}
      </Container>
    </>
  );
};
