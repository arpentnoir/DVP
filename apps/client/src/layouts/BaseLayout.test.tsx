import { render, waitFor } from '@testing-library/react';
import { BaseLayout } from './BaseLayout';

describe('BaseLayout', () => {
  it('should render correctly', () => {
    const { baseElement } = render(
      <BaseLayout title="Hello world">
        <></>
      </BaseLayout>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should show page title', async () => {
    render(
      <BaseLayout title="Hello world">
        <p>Hello</p>
      </BaseLayout>
    );

    await waitFor(() =>
      expect(global.document.title).toStrictEqual('Hello world')
    );
  });
});
