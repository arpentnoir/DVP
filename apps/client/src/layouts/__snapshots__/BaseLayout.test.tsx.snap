// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLayout should render correctly 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root MuiContainer-disableGutters css-1y9y4eh-MuiContainer-root"
    />
  </div>
</body>
`;
